syncd/VirtualOidTranslator.cpp

```diff
@@ -22,6 +22,28 @@ VirtualOidTranslator::VirtualOidTranslator(
     // empty
 }
 
+bool VirtualOidTranslator::tryTranslateRidToVid(
+        _In_ sai_object_id_t rid,
+        _Out_ sai_object_id_t &vid)
+{
+    SWSS_LOG_ENTER();
+
+    std::lock_guard<std::mutex> lock(m_mutex);
+
+    if (rid == SAI_NULL_OBJECT_ID)
+        return false;
+
+    if (m_rid2vid.find(rid) == m_rid2vid.end())
```

Comment:
condition is wrong here, if rid is not found in local map, it can still be in m_clinet, so reverse this, if rid is found, return true, and if rid is not found, then check m_client

Reply:
if (rid == SAI_NULL_OBJECT_ID)
    {
        SWSS_LOG_DEBUG("translated RID null to VID null");

        vid = SAI_NULL_OBJECT_ID;
        return true;
    }

    auto it = m_rid2vid.find(vid);

    if (it != m_rid2vid.end())
    {
        vid = it->second;
        return true;
    }

    vid = m_client->getVidForRid(rid);

    if(vid == SAI_NULL_OBJECT_ID)
    {
        SWSS_LOG_DEBUG("translated RID %s to VID null", sai_serialize_object_id(rid).c_str());
        return false;
    }

    return true;

I s this correction fine?

Reply:
looks good, also you write the code so you should test this scenario :P

Reply:
UT log:

root@sonic:~# show log process_on_port_state_change
Dec  7 02:40:39.168385 sonic DEBUG syncd#syncd: :> process_on_port_state_change: enter
Dec  7 02:40:39.168409 sonic DEBUG syncd#syncd: :- process_on_port_state_change: port notification count: 1
Dec  7 02:40:39.168442 sonic DEBUG syncd#syncd: :- process_on_port_state_change: Port RID oid:0x10000006b state change notification
Dec  7 02:40:39.168626 sonic INFO syncd#syncd: :- process_on_port_state_change: Port VID oid:0x1000000000034 state change notification
Dec  7 02:40:39.169020 sonic DEBUG syncd#syncd: :< process_on_port_state_change: exit
root@sonic:~# redis-cli -n 1 hgetall RIDTOVID > rid2vid.txt
root@sonic:~# grep -n 0x10000006b rid2vid.txt 
1081:oid:0x10000006b
root@sonic:~# grep -n 0x1000000000034 rid2vid.txt 
1082:oid:0x1000000000034
root@sonic:~#